library_nativeiofs.js

```diff
@@ -184,9 +188,29 @@ mergeInto(LibraryManager.library, {
 
       setattr: function(node, attr) {
         NATIVEIOFS.debug('setattr', arguments);
-        //TODO: implement once setLength lands
-        console.log('NATIVEIOFS error: setattr is not implemented, but continuing',
-                      'without throwing');
+        if ('size' in attr) {
+          let useOpen = false;
+          let handle = node.handle;
+          try {
+            if (!handle) {
+	      // Open a handle that is closed later.
+              useOpen = true;
+              handle = nativeIO.openSync(NATIVEIOFS.encodedPath(node));
+            }
+	    handle.setLength(attr.size);
+          } catch (e) {
+            if (!('code' in e)) throw e;
+            throw new FS.ErrnoError(-e.errno);
+          } finally {
+            if (useOpen && handle) {
+              handle.close();
+            }
+          }
+        }
+	else {
+	  console.log('NATIVEIOFS error: setattr is not implemented, but continuing',
```

Comment:
I think we can go ahead and support the other possible parameters? From https://github.com/emscripten-core/emscripten/blob/master/src/library_memfs.js#L190, it looks like the other two are just setting a node property in our case.